_posts/2020-08-11-backports-application.adoc

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: From zero to hero on backporting pull requests
+date: 2020-08-11
+tags: user-story
+author: gastaldi
+---
+= From zero to hero on backporting pull requests
+
+The Quarkus project moves fast and when we prepare bugfix releases, we usually have several dozens of pull requests to backport. Using the GitHub UI to do that is feasible but cumbersome.
+
+Before we dig into the solution, let me give you a quick explanation about our release process.
+
+== Release process
+
+The Quarkus Team adopts a `Major.Minor.Patch.Classifier` (eg. _1.7.0.CR1_, _1.7.0.Final_) version pattern. Depending on the version bumped, different processes are adopted:
+
+=== Major and Minor bumps
+Our master branch is always ready for the next major or minor release.  This process is usually very smooth and involves no backporting.
+
+=== Patch or Classifier bumps
+Whenever a new patched version (or a second CR) is about to be released in Quarkus, our Release team starts backporting commits from pull requests that were merged in the master branch.
+How do they know which pull request to grab? We have a `triage/backport?` label that our team add to pull requests for features that would be worth having it in the upcoming patch release.
+
+Considering the amount of backports, this gets tedious and error prone when done manually. So we decided to automate that, and built an application. Of course, with Quarkus!
+
+== What the application does?
+The application basically queries a GitHub repository's merged pull requests and closed issues (using https://docs.github.com/en/graphql[the GraphQL API exposed by GitHub]) containing a certain label and changes their milestones to one that is selected in the UI (and removing the certain label afterwards)
+
+=== Why we needed it?
+The number of pull requests to backport is enormous and (as it was aforementioned) doing it via the GitHub UI is inconvenient and takes a lot of time (click PR, copy/paste commit hashes to cherry-pick, remove label, affect milestone, affect milestone to all fixes issues, next) plus some limitation of the UI (it's not possible to sort by merge date to avoid conflicts, for example).
+Applying the changes introduced by these pull-requests (aka. cherry-picking) is also simplified.
```

Comment:
OK, so does it just simplify working with the issues, or does it automate the backporting process itself, too?

Reply:
Just the issues, the backporting process is done manually done. I've changed the sentence a bit